dali/operators/reader/nvdecoder/sequencewrapper.h

```diff
@@ -47,44 +47,23 @@ struct SequenceWrapper {
     timestamps.clear();
     timestamps.reserve(count);
 
-    int dev;
-    CUDA_CALL(cudaGetDevice(&dev));
-    std::unique_lock<std::mutex> lock{started_lock_};
-    if (started_) {
-      CUDA_CALL(cudaEventDestroy(event_));
-    }
-    CUDA_CALL(cudaEventCreateWithFlags(&event_, cudaEventBlockingSync | cudaEventDisableTiming));
-    started_ = false;
-  }
-
-  ~SequenceWrapper() {
-    std::unique_lock<std::mutex> lock{started_lock_};
-    if (started_) {
-      try {
-        CUDA_CALL(cudaEventDestroy(event_));
-      } catch (const std::exception &) {
-        // Something went wrong with releasing resources. We'd better die now.
-        std::terminate();
-      }
+    if (!event_) {
+      int dev;
+      CUDA_CALL(cudaGetDevice(&dev));
+      event_ = CUDAEvent::CreateWithFlags(cudaEventBlockingSync | cudaEventDisableTiming, dev);
```

Comment:
```suggestion
      event_ = CUDAEvent::CreateWithFlags(cudaEventBlockingSync | cudaEventDisableTiming);
```
This will automatically create an event for current device.

Reply:
Done